aioamqp/tests/test_channel.py

```diff
@@ -29,6 +29,16 @@ def test_close(self):
         self.assertFalse(channel.is_open)
 
     @testing.coroutine
+    def test_server_initiated_close(self):
```

Comment:
Ah, I guess the assertion is quite implicit. The bug this fixes stems from channel ID reuse.

If the server closes the channel it will ignore any messages about that channel ID until we ACK the close.
So when the new channel is created at the end of the test, it'll pick the same ID again, and if we didn't already ACK the close the server will ignore the messages about the new channel.
So I guess we check that we called it by virtue of the new channel working.
@foolswood: Was it enough to simply create the new channel to cause the hang, or did we have to try to publish through it too?

We could explicitly check that we send the message, do you have other tests which make that kind of assertion?
Another option: we could add some comments explaining the protocol assumptions this test is relying on, and add an assertion to check the channel IDs were the same (so we don't get false-positives if the ID picking logic is changed).